From 69bd76f2b90cd00c1596b2e2c05845a4d9596fd2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Sun, 6 Jan 2019 17:37:00 +0100
Subject: [PATCH] pam_systemd: set $DBUS_SESSION_BUS_ADDRESS unconditionally

There's very little lost if the variable is set for a socket that isn't
connectible, but a lot lost (races, ...) if it's not set but the socket exists.

Also, drop the FIXME note, since we don't plan to revert this revert any time
soon.
---
 src/login/pam_systemd.c | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/src/login/pam_systemd.c b/src/login/pam_systemd.c
index cdec102ce..d466e5096 100644
--- a/src/login/pam_systemd.c
+++ b/src/login/pam_systemd.c
@@ -198,18 +198,6 @@ static int export_legacy_dbus_address(
         _cleanup_free_ char *s = NULL;
         int r = PAM_BUF_ERR;
 
-        /* FIXME: We *really* should move the access() check into the
-         * daemons that spawn dbus-daemon, instead of forcing
-         * DBUS_SESSION_BUS_ADDRESS= here. */
-
-        s = strjoin(runtime, "/bus");
-        if (!s)
-                goto error;
-
-        if (access(s, F_OK) < 0)
-                return PAM_SUCCESS;
-
-        s = mfree(s);
         if (asprintf(&s, DEFAULT_USER_BUS_ADDRESS_FMT, runtime) < 0)
                 goto error;
 
-- 
2.19.1

